### PR TITLE
Add name flag to kedro new

### PR DESCRIPTION
**Pull request by lrcouto**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description

Adds a `--name` flag to `kedro new`, allowing the user to skip the prompt that asks for a project name.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: kedro/framework/cli/starters.py [hunk 1]
```diff
@@ -125,6 +125,8 @@ class KedroStarterSpec:  # noqa: too-few-public-methods
     "6": "Pyspark",
 }
 
+NAME_ARG_HELP = "The name of the Kedro project."
```
Comment (merelcht) [inline kedro/framework/cli/starters.py:128]:
```suggestion
NAME_ARG_HELP = "The name of your new Kedro project."
```

File: kedro/framework/cli/starters.py [hunk 2]
```diff
@@ -321,6 +328,7 @@ def new(config_path, starter_alias, selected_addons, checkout, directory, **kwar
         config = _fetch_config_from_user_prompts(prompts_required, cookiecutter_context)
 
     config = _get_addons_from_cli_input(selected_addons, config)
+    config = _get_name_from_cli_input(name, config)
```
Comment (merelcht) [inline kedro/framework/cli/starters.py:331]: It's generally not good practice to modify your input, in this case `config`, in this way. Here it's also getting hard to follow what "config" is exactly. I'd probably change this to something like: 

```suggestion
    config_with_add_ons = _get_addons_from_cli_input(selected_addons, config)
    config_with_add_ons_and_name = _get_name_from_cli_input(name, config_with_add_ons)
```

File: kedro/framework/cli/starters.py [hunk 3]
```diff
@@ -258,7 +260,10 @@ def create_cli():  # pragma: no cover
 @click.option("--checkout", help=CHECKOUT_ARG_HELP)
 @click.option("--directory", help=DIRECTORY_ARG_HELP)
 @click.option("--addons", "-a", "selected_addons", help=ADDON_ARG_HELP)
-def new(config_path, starter_alias, selected_addons, checkout, directory, **kwargs):
+@click.option("--name", "-n", "name", help=NAME_ARG_HELP)
+def new(  # noqa: too-many-arguments
+    config_path, starter_alias, selected_addons, name, checkout, directory, **kwargs
```
Comment (merelcht) [inline kedro/framework/cli/starters.py:265]: I didn't notice that `selected_addons` wasn't added as the last argument before `**kwargs` earlier on, but usually new arguments should be added at the end, because adding them somewhere randomly is a breaking change. Of course, all these changes are breaking anyway, but it's good to keep this in mind.

**Review by merelcht — COMMENTED** [review]
I've left some initial comments. I'll try out the code and come back to reviewing 🙂

**Comment by lrcouto** [comment]
@DimedS @merelcht that does make sense. I'll see if I can find a better solution for that.

**Comment by lrcouto** [comment]
Extracted the part of `kedro new` that builds the config to another function, so we can avoid reassigning it so many times.

File: kedro/framework/cli/starters.py [hunk 4]
```diff
@@ -359,20 +367,49 @@ def list_starters():
...
-    """Inserts add-on selection from the CLI input in the project
-    configuration, if it exists. Replaces add-on strings with the
-    corresponding prompt number.
+def _get_config(
+    prompts_required,
+    config_path,
+    cookiecutter_context,
+    selected_addons,
+    project_name,
+):
```
Comment (merelcht) [inline kedro/framework/cli/starters.py:376]: I think it would be good to have a doc string here, explaining what the method does.

**Review by merelcht — APPROVED** [review]
I left one minor comment and this change should also be added to the release notes, but apart from that this looks good to be merged 👍

**Comment by DimedS** [comment]
Looks like we have an error now with 'kedro new' choosing then 'none' option, but it seems not to be because of that PR, because the same is in the Develop branch, but maybe we should fix:

Traceback (most recent call last):
  File "/var/folders/j8/152bxdnd0tzbnx9qzklgmc040000gp/T/tmpb1mbj2ce.py", line 26, in <module>
    main()
  File "/var/folders/j8/152bxdnd0tzbnx9qzklgmc040000gp/T/tmpb1mbj2ce.py", line 23, in main
    sort_requirements(requirements_file_path)
  File "/Users/Dmitry_Sorokin/miniconda3/envs/kedroName3/lib/python3.9/site-packages/kedro/templates/project/hooks/utils.py", line 139, in sort_requirements
    with open(requirements_file_path, 'r') as requirements:
FileNotFoundError: [Errno 2] No such file or directory: '/Users/Dmitry_Sorokin/Git/Projects/dimed6/requirements.txt'
[11/01/23 14:40:58] ERROR    Stopping generation because post_gen_project hook script didn't exit successfully            hooks.py:159
kedro.framework.cli.utils.KedroCliError: Failed to generate project when running cookiecutter.
Run with --verbose to see the full exception
Error: Failed to generate project when running cookiecutter.

**Comment by lrcouto** [comment]
> Looks like we have an error now with 'kedro new' choosing then 'none' option, but it seems not to be because of that PR, because the same is in the Develop branch, but maybe we should fix:
> 
> Traceback (most recent call last): File "/var/folders/j8/152bxdnd0tzbnx9qzklgmc040000gp/T/tmpb1mbj2ce.py", line 26, in main() File "/var/folders/j8/152bxdnd0tzbnx9qzklgmc040000gp/T/tmpb1mbj2ce.py", line 23, in main sort_requirements(requirements_file_path) File "/Users/Dmitry_Sorokin/miniconda3/envs/kedroName3/lib/python3.9/site-packages/kedro/templates/project/hooks/utils.py", line 139, in sort_requirements with open(requirements_file_path, 'r') as requirements: FileNotFoundError: [Errno 2] No such file or directory: '/Users/Dmitry_Sorokin/Git/Projects/dimed6/requirements.txt' [11/01/23 14:40:58] ERROR Stopping generation because post_gen_project hook script didn't exit successfully hooks.py:159 kedro.framework.cli.utils.KedroCliError: Failed to generate project when running cookiecutter. Run with --verbose to see the full exception Error: Failed to generate project when running cookiecutter.

For `kedro new --addons=none`? I just ran it here with no issues. 🤔

**Comment by merelcht** [comment]
@DimedS What command did you run that gave you the error? I've also tried several combinations of `kedro new --addons=none` `kedro new --addons=none --name=myproject` and it runs without issues. cc: @lrcouto

**Comment by DimedS** [comment]
> @DimedS What command did you run that gave you the error? I've also tried several combinations of `kedro new --addons=none` `kedro new --addons=none --name=myproject` and it runs without issues. cc: @lrcouto

that's funny because that error was demonstrated by @SajidAlamQB in today's Tech Design meeting) I forgot to ask him what he did with that)

**Review by DimedS — APPROVED** [review]
@lrcouto thank you, looks good to me!

**Comment by lrcouto** [comment]
> > @DimedS What command did you run that gave you the error? I've also tried several combinations of `kedro new --addons=none` `kedro new --addons=none --name=myproject` and it runs without issues. cc: @lrcouto
> 
> that's funny because that error was demonstrated by @SajidAlamQB in today's Tech Design meeting) I forgot to ask him what he did with that)

I think that error was because he already had a directory with the same name as the one he was trying to create, IIRC

File: kedro/framework/cli/starters.py [hunk 5]
```diff
@@ -429,6 +487,16 @@ def _select_prompts_to_display(prompts_required: dict, selected_addons: str) ->
             sys.exit(1)
         del prompts_required["add_ons"]
 
+    if project_name is not None:
+        if bool(re.match(r"^[\w -]{2,}$", project_name)) is False:
```
Comment (noklam) [inline kedro/framework/cli/starters.py:491]: This could be written as 
```not re.match(r"^[\w -]{2,}$", project_name)```
